### ISSUE TITLE
flaskr project (in dev docs and on github) needs to use relative import in __init__.py for Python 3

### ISSUE THREAD

**Issue by toddjames** [pallets/flask#2051]
Currently, while following the dev docs or cloning the existing flaskr project on GitHub, it will not run in a Python 3 virtualenv. The error when running the cloned flaskr application is:

`Traceback (most recent call last):
  File "/home/todd/flaskr/flask/examples/flaskr/env/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/home/todd/flaskr/flask/examples/flaskr/flaskr/__init__.py", line 1, in <module>
    from flaskr import app
ImportError: cannot import name 'app'`

This is the same error when following the docs to recreate the flaskr application. By modifying `flaskr/__init__.py` to include a relative import when using a Python 3 virtualenv, everything works fine.

That is, in `flaskr/__init__.py`, change:
`from flaskr import app`
to:
`from .flaskr import app`

(note the period)

At this point, I am still not comfortable contributing, nor do I know if this is the correct solution or if this has any implications for Python 2. Hence the issue, hoping someone else can confirm this is correct, or provide a better solution.

**Comment by wgwz** [comment]
Good catch, this is my mistake. That line should be `from flaskr.flaskr import app`. Does this import statement fix it for you?

**Comment by toddjames** [comment]
Yes, that also fixes it.

**Comment by toddjames** [comment]
Thanks for the quick fix to the project itself. You may want to fix the documentation on this page, too:

http://flask.pocoo.org/docs/dev/tutorial/packaging/#tutorial-packaging

**Comment by untitaker** [comment]
That should *eventually* update automatically.

**Comment by wgwz** [comment]
@untitaker Are there tools available to link parts of a code example to documentation? That would be quite nice.

**Comment by untitaker** [comment]
What do you mean?

**Comment by untitaker** [comment]
Are you talking about the examples directory in the repo, or inline code examples?

**Comment by georgeberry** [comment]
I found this error as well. This is my `__init__.py` file:

```
from flask import Flask
app = Flask(__name__)

from . import views # relative import to fix github issue #2058
```
